Committing the changes for data model binding.

ashish-compro opened the pull request.

ashish-compro committed:
Committing the changes for data model binding.

ashish-compro commented on client/app/shared/text-box/text-box.component.ts:
```diff
@@ -10,12 +10,13 @@ import { itemSchema, itemDataModel } from '../UIConfig.model';
+  modelRef = { value: "" };
 
   ngOnInit() {
     this.itemDataModel = itemDataModel;
```
Need to remove the if check as there will be no case when we won't have the model reference for text box component.

ashish-compro commented on client/app/step-builder/shared/builder-model.service.ts:
```diff
@@ -0,0 +1,80 @@
+    }
+
+    getModelRef(modelRefStr) {
+        return eval("this.builderModel." + modelRefStr.replace(/['{','}']/g, ""));
```
Change the tokens from '{' to '{{'.
Rename the function name to getCompModel

ashish-compro committed:
Cherry picking the class inheritance correction commit and resolving the conflicts.

ashish-compro committed:
Committing the changes corresponding to the feedback.

ashish-compro committed:
Committing the enhancements for the approach.

prateekbhutani committed:
Requiring Translator Functions.

1. Changed Skill Manager from service to Class
2. Relocated Skill Manager to Shared folder
3. Accessing the Skill Translator Object of the skill Manager Class in Base Component

prateekbhutani committed:
Renaming Class

prateekbhutani committed:
Merge pull request #31 from Compro-Single-Step/Prateek/fdb-dependencyupdate

Requiring Translator Functions.

aman-compro committed:
use ENUM for label type values

aman-compro committed:
move default height and width of dropzone to scss

ashish-compro committed:
Merge pull request #35 from Compro-Single-Step/labelCompEnum

use ENUM for label type values

ashish-compro committed:
Merge pull request #36 from Compro-Single-Step/heightwidthdropzone

moved default height and width of dropzone to scss

aman-compro committed:
change app-wrapper to app-tab-page in tab component scss

ashish-compro committed:
implementing the data binding for rest of the components.

vsingb3 commented on server/controllers/db.controller.js:
```diff
@@ -45,6 +45,6 @@ function getFilePath(templateId, fileKey, callback) {
 
     skillConfigRepoModel.getFilePath({"template_id": templateId}, filterCriteria, (error, data) => {
-        callback(data[0][fileKey], error);
+        callback("filestore/skills/"+data[0][fileKey], error);
```
Not sure about this.

vsingb3 reviewed (approved):
Approving to enable resolve conflict option

ashish-compro committed:
Merge branch 'develop' into frontend-data-binding

Resolving Conflicts:
	client/app/shared/base.component.ts
	client/app/shared/dropzone/dropzone.component.ts
	client/app/shared/panel/panel.component.ts
	client/app/shared/tab/tab.component.ts
	client/app/shared/text-box/text-box.component.ts
	client/app/shared/wrapper/wrapper.component.ts
	client/app/step-builder/shared/builder-data.service.ts
	server/controllers/db.controller.js
	server/fileStore/skills/xl/MoveCellContent/uiconfig.json
	server/libs/skills/xl/moveCellContent/moveCellContent.js

ashish-compro committed:
Merge branch 'develop' into frontend-data-binding

Resolving Conflicts:
	server/libs/skills/xl/moveCellContent/moveCellContent.js

ashish-compro committed:
minor changes

ashish-compro committed:
Adding code for binding the options of a dropdown with the model.

prateekbhutani committed:
Getting Skill Translator
Minor update in UI Config
Updating Skill Translator Functions using Added Script

prateekbhutani committed:
Committing the changes for data binding and dynamic updation.

prateekbhutani committed:
Minor UI Config Update

No Regression.

ashish-compro committed:
Code changes required for supporting the dynamic updation in the third view.

ashish-compro committed:
Cleaning the code.

Commit:
Merge branch 'develop' into frontend-data-binding
